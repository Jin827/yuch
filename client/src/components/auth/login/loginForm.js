import React from 'react';
import { Link } from 'react-router-dom';
import FormControlLabel from '@material-ui/core/FormControlLabel';
import Checkbox from '@material-ui/core/Checkbox';
import { Formik, Form } from 'formik';
/* --- Components --- */
import FormikField from '../../../shared/form/formikField';
import PasswordField from '../../../shared/form/passwordField';
import Button from '../../../shared/form/formButton';

const LoginForm = ({
  keepMeLoggedIn,
  keepUserLoggedIn,
  userData,
  handleSubmit,
  loginValidation,
}) => {
  // when user found username from 'forgot ID', userData[0].username state is saved temporary on 'HTTP_SUCCESS' action of 'forgotUsername'.
  const foundUsername = userData.length !== 0 ? userData[0].username : '';
  const values = { username: foundUsername, password: '' };
  return (
    <Formik
      initialValues={values}
      validationSchema={loginValidation}
      onSubmit={handleSubmit}
    >
      {({ isSubmitting }) => (
        <div className="login-container">
          <Form className="flex flex-column-m items-center">
            <FormikField
              label="아이디"
              name="username"
              type="text"
              icon="filledUser"
              styleName="textField"
              required
            />
            <PasswordField
              label="비밀번호"
              name="password"
              styleName="textField"
            />
            <Button
              typeValue="submit"
              variantValue="contained"
              buttonName="로그인"
              className="login-btn"
              isSubmitting={isSubmitting}
            />
          </Form>
          <div className="keep-me-logged-in">
            <FormControlLabel
              control={
                <Checkbox
                  data-testid="checkbox-login"
                  checked={keepUserLoggedIn}
                  onClick={keepMeLoggedIn}
                  value="keepUserLoggedIn"
                />
              }
              label="로그인 상태 유지"
            />
          </div>
          <div className="tc login-forgot">
            <Link
              to="/auth/forgot?value=username"
              className="td-none c-text-grey"
            >
              아이디 찾기
            </Link>
            &#8201;&#8201;&#124;&#8201;&#8201;
            <Link
              to="/auth/forgot?value=password"
              className="td-none c-text-grey"
            >
              비밀번호 찾기
            </Link>
          </div>
        </div>
      )}
    </Formik>
  );
};

export default LoginForm;
