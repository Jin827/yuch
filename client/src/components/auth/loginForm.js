import React from 'react';
import TextField from '@material-ui/core/TextField';

const LoginForm = ({ handleChange, id, password }) => (
  <div className="mh1">
    <TextField
      id="id"
      label="ID"
      value={id}
      onChange={ev => handleChange(ev)}
      margin="normal"
      fullWidth
    />
    <TextField
      id="password"
      label="Password"
      value={password}
      onChange={ev => handleChange(ev)}
      margin="normal"
      fullWidth
    />
  </div>
);

export default LoginForm;
