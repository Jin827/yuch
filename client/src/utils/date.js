import moment from 'moment';
import { today } from '../helpers/moment';

export const dayBefore = date =>
  moment(date)
    .add(-1, 'days')
    .format('YYYYMMDD');

export const dayAfter = date =>
  moment(date)
    .add(1, 'days')
    .format('YYYYMMDD');

export const weekBefore = date =>
  moment(date)
    .add(-7, 'days')
    .format('YYYYMMDD');

export const weekAfter = date =>
  moment(date)
    .add(7, 'days')
    .format('YYYYMMDD');

export const monthBefore = date =>
  moment(date)
    .add(-1, 'months')
    .format('YYYYMMDD');

export const monthAfter = date =>
  moment(date)
    .add(1, 'months')
    .format('YYYYMMDD');

export const yearBefore = date =>
  moment(date)
    .add(-1, 'years')
    .format('YYYYMMDD');

export const yearAfter = date =>
  moment(date)
    .add(1, 'years')
    .format('YYYYMMDD');

const lastMonth = moment()
  .add(-1, 'months')
  .format('YYYYMM');

export const firstDayOfLastMonth = () => {
  const value = `${lastMonth}01`;
  return value;
};

export const formatToDateForm = date =>
  moment(date).format('MM 월 DD 일 (ddd)');

export const formatToMonthDateForm = date =>
  moment(date, 'YYYYMMDD').format('YYYY 년 MM 월');

export const formatToYearDateForm = date =>
  moment(date, 'YYYYMMDD').format('YYYY 년');

export const formatToYYYY = date => moment(date).format('YYYY');
export const formatToYYYYMM = date => moment(date).format('YYYYMM');
export const formatToYYYYMMDD = date => moment(date).format('YYYYMMDD');
export const formatWithDash = date => moment(date).format('YYYY-MM-DD');
export const formatWithSlash = date => moment(date).format('YYYY/MM/DD');

export const invoiceFormat = value => {
  const date = moment(value).format('DD');
  const day = moment(value).format('ddd');
  return { date, day };
};
export const formatSlashToYYMM = date =>
  moment(date, 'YYYY/MM').format('YYYYMM');

export const revenueFormat = date => moment(date, 'YYYYMM').format('MM');

export const isLunchQtyChangeDisabled = date => {
  if (date >= today) {
    const timeStamp = moment();
    const endTime = moment(`${date} 0930`, 'YYYYMMDD hhmm');

    if (timeStamp.isBefore(endTime)) {
      return false;
    }
    return true;
  }
  return true;
};

export const isDinnerQtyChangeDisabled = date => {
  if (date >= today) {
    const timeStamp = moment();
    const endTime = moment(`${date} 1330`, 'YYYYMMDD hhmm');

    if (timeStamp.isBefore(endTime)) {
      return false;
    }
    return true;
  }
  return true;
};

export const selectOptionsYYYYMM = arr =>
  arr.map(i => {
    const date = moment()
      .add(i, 'months')
      .format('YYYY/MM');
    return { value: date };
  });

export const userEndDate = (data, limitedDate) => {
  let endDate;
  if (data && data.endDate <= limitedDate) {
    endDate = data.endDate;
  } else {
    endDate = limitedDate;
  }
  return endDate;
};

export const isActiveUser = (startDate, endDate) => {
  const formattedStartDate = formatWithDash(startDate);
  const today = moment().format('YYYY-MM-DD');
  const isActive = moment(today).isBetween(formattedStartDate, endDate, 'days');
  return isActive;
};
