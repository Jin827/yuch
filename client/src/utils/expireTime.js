import moment from 'moment';

export const calculateExpireTime = expiresIn => {
  const expireTime = moment()
    .add(parseInt(expiresIn, 10), 'milliseconds')
    .format('YYYYMMDDhhmm');
  return expireTime;
};
