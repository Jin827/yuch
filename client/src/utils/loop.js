export const createForLoopArray = async (num, name) => {
  const promises = [];
  for (let i = 0; i < num; i++) {
    promises.push(`${name}${i}`);
  }
  const arr = await Promise.all(promises);
  return arr;
};
