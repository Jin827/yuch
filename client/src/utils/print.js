export const printDiv = async divName => {
  console.log('divName: ', divName);
  const printContents = document.getElementById(divName).innerHTML;
  const originalContents = document.body.innerHTML;
  document.body.innerHTML = printContents;
  window.print();
  document.body.innerHTML = originalContents;

  // const mywindow = await window.open('', 'PRINT', 'height=400,width=600');
  // mywindow.document.body.innerHTML = await printContents;
  // mywindow.document.close(); // necessary for IE >= 10
  // mywindow.focus(); // necessary for IE >= 10*/
  // mywindow.print();
  // mywindow.close();

  // const mywindow = window.open('', 'PRINT', 'height=400,width=600');
  // mywindow.document.write(`<html><head><title>${document.title}</title>`);
  // mywindow.document.write(
  //   '<link rel="stylesheet" href="../../styles/main.scss" type="text/css" />',
  // );
  // mywindow.document.write('</head><body >');
  // mywindow.document.write(`<h1>${document.title}</h1>`);
  // mywindow.document.write(printContents);
  // mywindow.document.write('</body></html>');
  // mywindow.document.close(); // necessary for IE >= 10
  // mywindow.focus(); // necessary for IE >= 10*/
  // mywindow.print();
  // mywindow.close();

  // return true;
};
