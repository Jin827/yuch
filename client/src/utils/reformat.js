export const ccyFormat = num => `${num.toFixed(2)}`;

export const formatNumber = num =>
  num.toString().replace(/(\d)(?=(\d{3})+(?!\d))/g, '$1,');

export const combinedFormat = num => {
  const cyyFormattedNum = ccyFormat(num);
  const formattedNum = formatNumber(cyyFormattedNum);
  return formattedNum;
};

export const removeRegExp = str => {
  const splitted = str.split('');
  const filtered = splitted.filter(i => i !== '-');
  const newStr = filtered.join('');
  return newStr;
};
