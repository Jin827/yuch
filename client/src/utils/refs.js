export const createInputArray = async num => {
  const promises = [];
  for (let i = 0; i < num; i++) {
    promises.push(`input${i}`);
  }
  const arr = await Promise.all(promises);
  return arr;
};
