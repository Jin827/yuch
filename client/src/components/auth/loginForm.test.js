import React from 'react';
import { shallow } from 'enzyme';
import { Unwrapped as UnwrappedLoginForm } from './loginForm';
import { findByTestAttr } from '../../utils/test';

const mockChange = jest.fn();
const mockSubmit = jest.fn();
const mockBlur = jest.fn();

const setUp = (props = {}) => {
  const component = shallow(<UnwrappedLoginForm {...props} />);
  return component;
};

describe('Login Form Component', () => {
  let wrapper;
  beforeEach(() => {
    const props = {
      values: { username: '', password: '' },
      errors: { username: false, password: false },
      touched: { username: false, password: false },
      handleChange: mockChange,
      handleSubmit: mockSubmit,
      isSubmitting: false,
      handleBlur: mockBlur,
      classes: {
        textField: '',
      },
    };

    wrapper = setUp(props);
  });

  it('It should render without errors', () => {
    expect(wrapper).toMatchSnapshot();
    const loginformComponent = findByTestAttr(wrapper, 'loginformComponent');
    expect(loginformComponent).toHaveLength(1);
  });

  it('It should render textfield and button', () => {
    const textfield = wrapper.find('TextField');
    expect(textfield).toHaveLength(2);

    const button = wrapper.find('WithStyles(FormButton)');
    expect(button).toHaveLength(1);
  });

  describe('It should show helperText and error when errors and touched are true', () => {
    beforeEach(() => {
      wrapper.setProps({
        errors: { username: true, password: true },
        touched: { username: true, password: true },
      });
    });

    it('username', () => {
      const textField = wrapper.find(`[data-test='username-input']`);
      // expect(textField.props().helperText).toEqual(true);
      expect(textField.props().error).toEqual(true);
    });
    it('password', () => {
      const textField = wrapper.find(`[data-test='password-input']`);
      // expect(textField.props().helperText).toEqual(true);
      expect(textField.props().error).toEqual(true);
    });
  });

  describe('It should NOT show helperText and error if one of errors and touched is not true', () => {
    beforeEach(() => {
      wrapper.setProps({
        errors: { username: false, password: true },
        touched: { username: true, password: false },
      });
    });

    it('username', () => {
      const textField = wrapper.find(`[data-test='username-input']`);
      // expect(textField.props().helperText).toEqual(false);
      expect(textField.props().error).toEqual(false);
    });
    it('password', () => {
      const textField = wrapper.find(`[data-test='password-input']`);
      // expect(textField.props().helperText).toEqual(false);
      expect(textField.props().error).toEqual(false);
    });
  });

  describe('If the value is false or undefined', () => {
    beforeEach(() => {
      wrapper.setProps({
        values: { username: false, password: undefined },
      });
    });

    it('Value should have an empty string', () => {
      const usernameTextField = findByTestAttr(wrapper, 'username-input');
      const passwordTextField = findByTestAttr(wrapper, 'password-input');
      expect(usernameTextField.props().value).toEqual('');
      expect(passwordTextField.props().value).toEqual('');
    });
  });

  describe('When submit button is clicked', () => {
    beforeEach(() => {
      // wrapper.find('form').simulate('submit', { preventDefault() {} });
    });
  });

  it('call submit function', () => {
    // expect(mockSubmit).toHaveBeenCalledTimes(1);
    // expect(submitBtn.prop(isSubmitting)).toBe(true);
  });
});
