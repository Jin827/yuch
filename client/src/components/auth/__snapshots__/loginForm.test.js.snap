// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Form Component It should render without errors 1`] = `
<div
  className="login-container"
  data-test="loginformComponent"
>
  <form
    className="flex flex-column-m items-center"
    onSubmit={[MockFunction]}
  >
    <TextField
      InputProps={
        Object {
          "startAdornment": <WithStyles(InputAdornment)
            position="start"
          >
            <Icon
              fill="none"
              height="30"
              name="filledUser"
              viewBox="0 0 30 30"
              width="30"
            />
          </WithStyles(InputAdornment)>,
        }
      }
      className=""
      data-test="username-input"
      error={false}
      helperText={false}
      id="username"
      label="Username"
      margin="normal"
      name="username"
      onBlur={[MockFunction]}
      onChange={[MockFunction]}
      required={true}
      select={false}
      value=""
      variant="standard"
    />
    <TextField
      InputProps={
        Object {
          "startAdornment": <WithStyles(InputAdornment)
            position="start"
          >
            <Icon
              fill="none"
              height="30"
              name="filledLock"
              viewBox="0 0 30 30"
              width="30"
            />
          </WithStyles(InputAdornment)>,
        }
      }
      className=""
      data-test="password-input"
      error={false}
      helperText={false}
      id="password"
      label="Password"
      margin="normal"
      name="password"
      onBlur={[MockFunction]}
      onChange={[MockFunction]}
      required={true}
      select={false}
      value=""
      variant="standard"
    />
    <WithStyles(FormButton)
      buttonName="로그인"
      data-test="submit-btn"
      isSubmitting={false}
      typeValue="submit"
      variantValue="contained"
      width="medium"
    />
  </form>
</div>
`;
